libtransmission/crypto-utils-ccrypto.cc

```diff
@@ -119,9 +119,16 @@ class Sha1Impl final : public tr_sha1
 
     void add(void const* data, size_t data_length) override
     {
+        uint8_t const* shaData = (uint8_t const*)data;
+        while (data_length > UINT32_MAX)
```

Comment:
No, it shouldn't be larger than size_t.
And since adding a condition on `sizeof(size_t)` would still make it "always true" or "always false" on some platforms, this piece of code is a candidate to ignore "condition is always false" warning with `#pragma GCC diagnostic ignored`.

But I won't do it for now, as I'm not on such platform and I'm not seeing the harmless warning.

Reply:
I added a macro... likely unnecessary verbosity/optimization, but now it should have no warnings.